### ISSUE TITLE
Feature Suggestions: `MongitaStore`

### ISSUE THREAD

**Issue by arosen93** [materialsproject/maggma#828]
https://github.com/scottrogowski/mongita

It would be great to have more file-based DB options in maggma. One such option (which I think @rkingsbury first told me about) is `mongita`. This is mostly a note to myself that it would be nice to implement this one day. Perhaps I'll come back to this.

**Comment by arosen93** [comment]
Edit: Looks like we are out of luck there for Mongita. I imagine the same for MontyDB based on the code.

> Mongita is an embedded database. It is not process-safe. When you have multiple clients, a traditional server/client database is the correct choice.

There's nothing in maggma specifically to resolve this, right @munrojm?

**Comment by rkingsbury** [comment]
Yeah, thanks for the suggestion @arosen93 . Definitely no objections to implementing a `MongitaStore` someday. I looked briefly at `mongita` as a substitute for `mongomock` to power `MemoryStore`, but I noted the following limitations:

> mongita` does not support many query operations including `$regex` or `$exists`. It also doesn't support `bulk_write` or `estimated_document_count` although those can be worked around.

**Comment by Andrew-S-Rosen** [comment]
https://github.com/openjournals/joss-reviews/issues/5995#issuecomment-1810759481

Worth jotting down FerretDB here, as highlighted by @utf above: https://www.ferretdb.com/
